File: BEmu_cpp/src/MarketDataRequest/ElementMarketBool.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/MarketDataRequest/ElementMarketBool.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "MarketDataRequest/ElementMarketBool.h"

namespace BEmu
{
	namespace MarketDataRequest
	{
		ElementMarketBool::ElementMarketBool(std::string name, bool value)
		{
			this->_name = name;
			this->_value = value;
		}

		Name ElementMarketBool::name() const
		{
			return Name(this->_name.c_str());
		}

		size_t ElementMarketBool::numValues() const
		{
			return 1;
		}

		size_t ElementMarketBool::numElements() const
		{
			return 0;
		}

		bool ElementMarketBool::isNull() const
		{
			return false;
		}

		bool ElementMarketBool::isArray() const
		{
			return false;
		}

		bool ElementMarketBool::isComplexType() const
		{
			return false;
		}

		bool ElementMarketBool::getValueAsBool(int index) const
		{
			if(index == 0)
				return this->_value;
			else
				throw elementPtrEx;
		}

		std::ostream& ElementMarketBool::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			this->prettyPrintHelper(stream, level, spacesPerLevel, this->_value ? "True" : "False");
			return stream;
		}

	}
}

File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/MarketDataRequest/ElementMarketBool.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.MarketDataRequest.ElementMarketBool.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.MarketDataRequest;

import com.bemu.BEmu.Name;
import com.bemu.BEmu.Schema;

public class ElementMarketBool extends ElementParent
{
	private final boolean _value;
	private final String _name;
	
	ElementMarketBool(String name, boolean value)
	{
		this._name = name;
		this._value = value;
	}
	
    public Name name() throws Exception
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.BOOL;
    }
    
    protected StringBuilder prettyPrint(int tabIndent) throws Exception
    {
    	return super.prettyPrintHelper(tabIndent, String.valueOf(this._value));
    }
    
    public Object getValue()
    {
    	return this._value;
    }
    
    public boolean getElementAsBool(String name)
    {
    	return this._value;
    }
}
